cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
-    module_name = pydantic_schema.stem
-    click.echo(f"Got file {pydantic_schema}, trying to import it...")
-    spec = importlib.util.spec_from_file_location(module_name, pydantic_schema)
-    module = importlib.util.module_from_spec(spec)  # type: ignore [arg-type]
-    sys.modules[module_name] = module
-    spec.loader.exec_module(module)  # type: ignore [union-attr]
+def to_gql(
+    schema_module: Path = typer.Argument(
+        _schema_module or ..., help="Pydantic schema to convert. Path to a .py file or Python dot notation "
+    ),
+    graphql_schema: Path = typer.Option(_graphql_schema or ..., help="File path for the output."),
+    prefix: str = typer.Option(_prefix, help="Name prefix for the domain."),
+):
+    if str(schema_module).endswith(".py"):
+        click.echo(f"Got file {schema_module}, trying to import it...")
+        module_name = schema_module.stem
+        spec = importlib.util.spec_from_file_location(module_name, schema_module)
+        module = importlib.util.module_from_spec(spec)  # type:ignore[arg-type]
+        sys.modules[module_name] = module
+        spec.loader.exec_module(module)  # type:ignore[union-attr]
+    else:
+        module_name = settings.local.get("schema_module", default=schema_module.stem)
+        click.echo(f"Got module {schema_module}, trying to import it...")
+        module = importlib.import_module(module_name)
     click.echo("Import successful")
 
-    click.echo("Searching for a schema...")
-    for schema_name, instance in inspect.getmembers(module):
-        if isinstance(instance, Schema):
-            click.echo(f"Found schema {schema_name!r}")
-            break
+    if not _prefix:
+        click.echo("Searching for a schema...")
+        for schema_name, instance in inspect.getmembers(module):
+            if isinstance(instance, Schema):
+                click.echo(f"Found schema {schema_name!r}")
+                break
+        else:
+            click.echo("Failed to find schema, exiting..")
+            exit(1)
     else:
-        click.echo("Failed to find schema, exiting..")
-        exit(1)
+        schema_name = to_schema_name(prefix)
+        click.echo(f"Got schema name {schema_name}")
+        instance = getattr(module, schema_name)
 
-    output = Path.cwd() / "schema.graphql"
-    output.write_text(instance.as_str())
-    click.echo(f"Wrote file {output}")
+    graphql_schema.write_text(instance.as_str())
+    click.echo(f"Wrote file {graphql_schema}")
+
+
+@app.command("signin", help="Upload a GQL schema to CDF DM data model.")
+def signin(
+    cdf_cluster: str = typer.Option(_cdf_cluster or ..., help="CDF cluster name."),
+    tenant_id: str = typer.Option(_tenant_id or ..., help="AD tenant ID."),
+    client_id: str = typer.Option(_client_id or ..., help="AD client ID."),
+    client_secret: str = typer.Option(
+        _client_secret_hidden or None, prompt=True, hide_input=True, help="AD client secret."
+    ),
```

Comment:
Agreed, those prompts were annoying. But on second look, we should really not read secrets form CLI arguments, that is insecure. I've changed it to take the secret either from settings or from a prompt, but it will prompt only it the settings value is empty. 
Also supports stdin, e.g. if the secret is in a file: `dm signing --other-options=... < file/with/the_secret`

https://github.com/cognitedata/cognite-gql-pygen/pull/20/commits/7595d34f42306f7f04edfd650cabdfa939294cf3